jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -186,6 +207,85 @@ async def job_script_update(job_submission_id: int, job_submission: JobSubmissio
     return job_submission_data
 
 
+# The "agent" routes are used for agents to fetch pending job submissions and update their statuses
+@router.get(
+    "/job-submissions/agent/pending",
+    description="Endpoint to list pending job_submissions for the requesting client",
+    response_model=List[PendingJobSubmission],
+)
+async def job_submissions_agent_pending(
+    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_SUBMISSIONS_VIEW)),
+):
+    """
+    Get a list of pending job submissions for the cluster-agent.
+    """
+    if token_payload.client_id is None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Access token does not contain a `client_id`. Cannot fetch pending submissions",
+        )
+
+    query = (
+        select(
+            columns=[
+                job_submissions_table.c.id,
+                job_submissions_table.c.job_submission_name,
+                job_scripts_table.c.job_script_name,
+                job_scripts_table.c.job_script_data_as_string,
+                applications_table.c.application_name,
+            ]
+        )
+        .select_from(job_submissions_table.join(job_scripts_table).join(applications_table))
+        .where(job_submissions_table.c.status == JobSubmissionStatus.CREATED,)
+        .where(job_submissions_table.c.cluster_client_id == token_payload.client_id,)
+    )
```

Comment:
this is interesting implementation. I wonder if we could improve it with GraphQL.. anyway, just thinking out loud here

Reply:
Unless we changed the data backend, we would still end up with something fundamentally like this. I think it might be worth exploring changing the api layer and the data layer at some point in the future.

Reply:
We will investigate this more in a future task.